lib/msal-node/src/client/ClientAssertion.ts

```diff
@@ -87,9 +87,9 @@ export class ClientAssertion {
         const issuedAt = TimeUtils.nowSeconds();
         this.expirationTime = issuedAt + 600;
 
-        const header = {
-            [JwtConstants.ALGORITHM]: JwtConstants.RSA_256,
-            [JwtConstants.X5T]: EncodingUtils.base64EncodeUrl(this.thumbprint, "hex")
+        const header: JwtHeader = {
+            alg: JwtConstants.RSA_256,
+            x5t: EncodingUtils.base64EncodeUrl(this.thumbprint, "hex")
```

Comment:
Typescript (at least the version I have running) was able to figure out that the constants used as dynamic keys here satisfied the type requirements (i.e. that `alg` is required), so changed to use the type from `jsonwebtoken`, which provides equivalent protection against misspelling the keys.